File: src/main/java/ru/glitchless/games/tprunner/utils/HashUtils.java
package ru.glitchless.games.tprunner.utils;

import java.io.BufferedInputStream;
import java.io.File;
import java.io.FileInputStream;
import java.security.MessageDigest;
import java.util.Base64;

public class HashUtils {
    public static String generateSHA256(File file) throws Exception {
        byte[] buffer = new byte[8192];
        int count;
        MessageDigest digest = MessageDigest.getInstance("SHA-256");
        BufferedInputStream bis = new BufferedInputStream(new FileInputStream(file));
        while ((count = bis.read(buffer)) > 0) {
            digest.update(buffer, 0, count);
        }
        bis.close();

        byte[] hash = digest.digest();
        return new String(Base64.getEncoder().encode(hash));
    }
}

File: src/jvmMain/kotlin/ru/lionzxy/techoparkrunner/utils/HashUtils.kt
package ru.lionzxy.techoparkrunner.utils

import java.security.MessageDigest
import kotlin.io.encoding.Base64
import kotlin.io.encoding.ExperimentalEncodingApi
import kotlin.math.sin
import okio.Buffer
import okio.FileSystem
import okio.HashingSink
import okio.Path
import okio.Sink
import okio.blackholeSink


@OptIn(ExperimentalEncodingApi::class)
object HashUtils {
    @Throws(Exception::class)
    fun generateSHA256(path: Path): String {
        val hash = HashingSink.sha256(blackholeSink()).use { sink ->
            FileSystem.SYSTEM.read(path) {
                this.readAll(sink)
            }
            sink.hash
        }
        return Base64.encode(hash.toByteArray())
    }
}

